src/components/ToggleGroup.tsx

```diff
@@ -31,35 +30,37 @@ const useStyles = makeStyles(() => ({
             "&:hover": {
                 backgroundColor: "rgba(0, 0, 0, 0.2)",
                 color: "rgba(0, 0, 0, 0.87)",
-                fill: "rgba(0, 0, 0, 0.87)"
-            }
+                fill: "rgba(0, 0, 0, 0.87)",
+            },
         },
-        "&>:first-child": {
+        "&>:first-of-type": {
             borderTopLeftRadius: "4px",
-            borderBottomLeftRadius: "4px"
+            borderBottomLeftRadius: "4px",
         },
         "&>:last-child": {
             borderTopRightRadius: "4px",
-            borderBottomRightRadius: "4px"
-        }
+            borderBottomRightRadius: "4px",
+        },
     },
     active: {
         backgroundColor: "rgba(0, 0, 0, 0.15)",
         color: "rgba(0, 0, 0, 0.87)",
-        fill: "rgba(0, 0, 0, 0.87)"
-    }
+        fill: "rgba(0, 0, 0, 0.87)",
+    },
 }));
 
 const ToggleGroup: React.FC<Props> = props => {
-    const classes = useStyles();
+    const { classes, cx } = useStyles();
+
     return (
-        <div className={clsx(classes.root, props.className)}>
+        <div className={cx(classes.root, props.className)}>
             {props.options.map(option => (
                 <button
                     key={option.id}
                     type="button"
-                    className={clsx(props.active === option.id && classes.active)}
-                    onClick={() => props.onChange(option.id)}>
+                    className={cx(props.active === option.id && classes.active)}
+                    onClick={() => void props.onChange(option.id)}
```

Comment:
Why do you expect a promise if you just ignore the result anyway? If you leave void, you should be able to return a Promise as well.

Reply:
I have to return a Promise because the callback function is async. If I remove void I get a eslint error:
`ESLint: Promise-returning function provided to attribute where a void return was expected.(@typescript-eslint/no-misused-promises)`